frontend/src/scenes/pipeline/pipelineAppLogsLogic.tsx

```diff
@@ -0,0 +1,240 @@
+                        values.searchTerm,
+                        logLevelsToTypeFilters(values.selectedLogLevels),
+                        null,
+                        values.leadingEntry as PluginLogEntry | null
+                    )
+                }
+
+                return [...results, ...values.backgroundLogs]
+            },
+        },
+    })),
+    reducers({
+        selectedLogLevels: [
+            Object.values(PipelineAppLogLevel).filter((level) => level !== 'DEBUG'),
+            {
+                setSelectedLogLevels: (_, { levels }) => levels,
+            },
+        ],
+        backgroundLogs: [
+            [] as PluginLogEntry[] | BatchExportLogEntry[],
+            {
+                clearBackgroundLogs: () => [],
+            },
+        ],
+        searchTerm: [
+            '',
+            {
+                setSearchTerm: (_, { searchTerm }) => searchTerm,
+            },
+        ],
+        isThereMoreToLoad: [
+            true,
+            {
+                loadLogsSuccess: (_, { logs }) => logs.length >= LOGS_PORTION_LIMIT,
+                markLogsEnd: () => false,
+            },
+        ],
+    }),
+    selectors({
+        leadingEntry: [
+            (s) => [s.logs, s.backgroundLogs],
+            (logs: LogEntry[], backgroundLogs: LogEntry[]): LogEntry | null => {
+                if (backgroundLogs.length) {
+                    return backgroundLogs[0]
+                }
+                if (logs.length) {
+                    return logs[0]
+                }
+                return null
+            },
+        ],
+        trailingEntry: [
+            (s) => [s.logs, s.backgroundLogs],
+            (logs: LogEntry[], backgroundLogs: LogEntry[]): LogEntry | null => {
+                if (logs.length) {
+                    return logs[logs.length - 1]
+                }
+                if (backgroundLogs.length) {
+                    return backgroundLogs[backgroundLogs.length - 1]
+                }
+                return null
+            },
+        ],
+        columns: [
+            (s) => [s.appType],
+            (appType): LemonTableColumns<LogEntry> => {
+                return [
+                    {
+                        title: 'Timestamp',
+                        key: 'timestamp',
+                        dataIndex: 'timestamp',
+                        render: (timestamp: string) => dayjs(timestamp).format('YYYY-MM-DD HH:mm:ss.SSS UTC'),
+                    },
+                    {
+                        title: appType === DestinationTypeKind.BatchExport ? 'Run Id' : 'Source',
+                        dataIndex: appType === DestinationTypeKind.BatchExport ? 'run_id' : 'source',
+                        key: appType === DestinationTypeKind.BatchExport ? 'run_id' : 'source',
+                    },
+                    {
+                        title: 'Level',
+                        key: appType === DestinationTypeKind.BatchExport ? 'level' : 'type',
```

Comment:
maybe would be nice if we did it the same way as with DestinationType - we could have PipelineAppLogType - then a selector, where we combine it from the two different options, which if there's a third thing latter and or any changes further down you'll only need to operate on a single type would be simpler 🤷